website/content/en/docs/building-operators/golang/advanced-topics.md

```diff
@@ -183,9 +183,8 @@ func (r *MemcachedReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 
     // Add finalizer for this CR
     if !contains(memcached.GetFinalizers(), memcachedFinalizer) {
-        if err := r.addFinalizer(reqLogger, memcached); err != nil {
-            return ctrl.Result{}, err
-        }
+        controllerutil.AddFinalizer(memcached, memcachedFinalizer)
+        r.Update(context.TODO(), memcached)
```

Comment:
`r.Update` returns an error. We'll want to check that. Similar to how we did it in `addFinalizer` method you removed.

```golang
    err := r.Update(context.TODO(), m)
    if err != nil {
        reqLogger.Error(err, "Failed to update Memcached with finalizer")
        return ctrl.Result{}, err
```

Reply:
:+1: Done and also noticed controllerutil.ContainsFinalizer is a thing too.